File: src/main/java/com/freya02/botcommands/api/waiter/CompletedFutureEvent.java
package com.freya02.botcommands.api.waiter;

import net.dv8tion.jda.api.events.GenericEvent;

import java.util.concurrent.Future;

/**
 * Just a TriConsumer for {@link EventWaiterBuilder#setOnComplete(CompletedFutureEvent)}, accepts a {@link Future} and the provided event or an exception<br>
 * <b>You will either receive the event object or a Throwable</b>
 *
 * @param <T> The JDA event waited for
 */
public interface CompletedFutureEvent<T extends GenericEvent> {
	void accept(Future<T> future, T e, Throwable t);
}

File: src/main/kotlin/com/freya02/botcommands/api/waiter/CompletedFutureEvent.kt
package com.freya02.botcommands.api.waiter

import net.dv8tion.jda.api.events.Event
import java.util.concurrent.Future

/**
 * Just a TriConsumer for [EventWaiterBuilder.setOnComplete],
 * provides a [Future] and **either** the event **or** an exception.
 *
 * @param T The JDA event waited for
 */
interface CompletedFutureEvent<T : Event> {
    fun accept(future: Future<T>, e: T?, t: Throwable?)
}
